heartbeat/Makefile

```diff
@@ -22,3 +22,8 @@ fields:
 	@mkdir -p _meta/
 	@cp ${ES_BEATS}/heartbeat/_meta/fields.common.yml _meta/fields.generated.yml
 	@cat ${ES_BEATS}/heartbeat/monitors/active/*/_meta/fields.yml >> _meta/fields.generated.yml
+
+# Create a new monitor. Requires the parameter MONITOR
+.PHONY: create-monitor
+create-monitor:
+	go run scripts/generate_monitor/main.go -monitor=${MONITOR}
```

Comment:
Nice, it's in golang. I wonder if we should put this under the `heartbeat/cmd/generate_monitor` package. It would follow better what we do with `dev-tools` but would conflict with the logic we have now that "all" commands under `cmd` can be accessed through `cobra`. @exekias Any thoughts?

Reply:
wouldn't put it in `heartbeat/cmd/generate_monitor`, as this would imply the code generator being available in the heartbeat binary itself. As the generator requires a full dev setup, I'd rather keep it where it is (well, I find having the generators in `scripts` somewhat confusing though).

Most of the go tool is driving the template engine. We can make it re-usable, but introducing some of the common functionalities in `beats/dev-tools/...`

Normally `cmd` indicates a multitude of compilable tools, but we use it for structuring sub-commands. Adding dev tools to `cmd`, I'd also move the current `cmd` package by one level to:

```
beats/heartbeat/cmd:
  generator/ ...  # the code generator
  main/ ...          # all contents currently in cmd package
```

or 

```
beats/heartbeat/cmd:
  generator/ ...  # the code generator
  heartbeat/ ...          # all contents currently in cmd package
```

Alternatively (given we take beat as framework more serious):

```
beats/cmd/
  filebeat/...
  packetbeat/...
  heartbeat/...
  generate.../...
beats/libbeat/...
beats/filebeat/...
beats/heartbeat/...
beats/packetbeat/...
```

Oh my

Reply:
Lets ignore that for now but I think we have similar ideas in mind moving forward.